### ISSUE TITLE
Positional Rotary Embeddings for transformers

### ISSUE THREAD

**Issue by TParcollet** [speechbrain/speechbrain#2677]
### Describe the bug

Samsung AI Center Cambridge (Shucong) may want to try to implement these embeddings for the v1.

### Expected behaviour

New feature.

**Comment by TParcollet** [comment]
@shucongzhang will take care of that.

**Comment by asumagic** [comment]
Somewhat related notes since I imagine this is also about performance:

`torch.nn.functional.scaled_dot_product_attention` apparently exists since 2.x (though is in "beta"), and it has an `attn_mask` parameter, which can behave as an additive bias to the attention scores (not just a boolean mask). Using a float attention bias might restrict which backends can be used, but maybe is a net benefit.  
I wonder if we can make a strictly conforming implementation of RelPosMHAXL using it.

Additionally, `torch.compile(..., dynamic=True)` seems to work and have some benefits on RelPosMHAXL on some GPUs, but my performance testing hasn't been extensive.
